packages/next/src/views/CreateFirstUser/CreateFirstUserForm/index.tsx

```diff
@@ -0,0 +1,100 @@
+'use client'
+import type { PayloadRequestWithData } from 'payload/types'
+
+export const createFirstUserBaseClass = 'create-first-user__form'
+
+import { ConfirmPassword } from '@payloadcms/ui/fields/ConfirmPassword'
+import { Email } from '@payloadcms/ui/fields/Email'
+import { Password } from '@payloadcms/ui/fields/Password'
+import { Form } from '@payloadcms/ui/forms/Form'
+import { RenderFields } from '@payloadcms/ui/forms/RenderFields'
+import { FormSubmit } from '@payloadcms/ui/forms/Submit'
+import { useComponentMap } from '@payloadcms/ui/providers/ComponentMap'
+import { useConfig } from '@payloadcms/ui/providers/Config'
+import { useTranslation } from '@payloadcms/ui/providers/Translation'
+import { email, password } from 'payload/fields/validations'
+import React from 'react'
+
+import './index.scss'
+
+export const CreateFirstUserForm: React.FC<{
+  searchParams: { [key: string]: string | string[] | undefined }
+}> = ({ searchParams }) => {
+  const { t } = useTranslation()
+
+  const config = useConfig()
+
+  const { getFieldMap } = useComponentMap()
+
+  const {
+    admin: { user: userSlug },
+    routes: { admin, api },
+  } = config
+
+  const fieldMap = getFieldMap({ collectionSlug: userSlug })
+
+  return (
+    <Form
+      action={`${api}/${userSlug}/first-register`}
+      method="POST"
+      redirect={typeof searchParams?.redirect === 'string' ? searchParams.redirect : admin}
+      validationOperation="create"
```

Comment:
Is there a reason you removed the onChange prop? Might cause things like field validation to not work anymore

Reply:
I don't think so. I was copying the same pattern from Login component and forgot to implement this variation of field validation. I'll fix that and ask you for review again. Thx.

Reply:
@AlessioGr I have a question about the codebase pattern. Should I use req in client components? Form component would require the "req" props as params to handle de formState
```
const initialState: FormState = await buildStateFromSchema({
    fieldSchema: fields,
    operation: 'create',
    preferences: { fields: {} },
    req,
  })
  ```

Reply:
moved to server side